src/main/kotlin/se/uulm/snowballr/backend/service/ProjectService.kt

```diff
@@ -194,4 +204,37 @@ class ProjectService(
             val projectMembersWithUsers = projectMemberRepo.getProjectMembersWithUsers(projectId)
             projectMembersWithUsers.toGrpcProjectMembers()
         }
+
+    override suspend fun getProjectInformation(request: GrpcProject.Information.Get): GrpcProject.Information =
+        withUser(userRepo) { currentUser ->
+            val projectId = parseUUID(request.projectId, EntityType.PROJECT)
+
+            isProjectExistent(repo)
+                .andAlso(isAllowedToReadProject(projectMemberRepo))
+                .checkFor(currentUser, projectId)
+
+            val project = repo.getProjectById(projectId).getOrThrow()
+            val progress = projectPaperRepo.getProjectProgress(projectId)
+            val builder = GrpcProject.Information.newBuilder()
+            val has = if (request.hasMask()) {
+                val fieldMaskPaths = FieldMaskUtil.normalize(request.mask).pathsList.toSet();
+                { path: String -> path in fieldMaskPaths }
+            } else {
+                { _ -> true }
+            }
```

Comment:
I think we should also add the following check to account for an empty path list:
```suggestion
            val has = if (request.hasMask() && request.mask.pathsList.isNotEmpty()) {
                val fieldMaskPaths = FieldMaskUtil.normalize(request.mask).pathsList.toSet();
                { path: String -> path in fieldMaskPaths }
            } else {
                { _ -> true }
            }
```

Payloads that return all fields:
```json
{
  "project_id": "05ec3232-7917-4819-a2e4-5289b1789c70"
}
```
```json
{
  "project_id": "05ec3232-7917-4819-a2e4-5289b1789c70",
  "mask": { }
}
```
```json
{
  "project_id": "05ec3232-7917-4819-a2e4-5289b1789c70",
  "mask": {
    "paths": [ ]
  }
}
```